wasmtime-wasi/src/host_impls.rs

```diff
@@ -121,7 +121,79 @@ macro_rules! rwlock_unlock {
     };
 }
 
-pub fn wasmtime_ssp_proc_exit(rval: wasm32::__wasi_exitcode_t) {
+pub fn wasmtime_ssp_args_get(
+    argv_environ: &mut host::argv_environ_values,
+    argv: &mut [*mut host::char],
+    argv_buf: &mut [host::char],
+) -> host::__wasi_errno_t {
+    for i in 0..(*argv_environ).argc {
+        let buf_off;
+        unsafe {
+            buf_off = usize::checked_sub(
+                *(*argv_environ).argv.offset(i as isize) as _,
+                (*argv_environ).argv_buf as _,
+            )
+            .expect("argv[i] - argv_buf overflows");
+        }
+        argv[i] = argv_buf[buf_off..].as_mut_ptr();
+    }
+    argv[(*argv_environ).argc] = std::ptr::null_mut();
+    unsafe {
+        argv_buf.copy_from_slice(slice::from_raw_parts_mut(
```

Comment:
Like below, `std::slice::copy_from_slice` is a safe fn, so you could take it out of the `unsafe` block, leaving only `std::slice::from_raw_parts_mut` which is unsafe.